src/template_generation_tools.py

```diff
@@ -100,6 +100,8 @@ def generate_class_graph_template(ccf_tools_df :pd.DataFrame):
 
   terms_ct_as_start = len(terms_ct_as)    
 
+  all_edges = ug.construct_uberon(" ".join(list(terms_pairs.union(terms_ct_as))), ug.construct_all_edges)
```

Comment:
Is this sufficient to generate all possible combinations of edges for testing in the where clause of construct_all_edges?  I'm wondering if this is why some edges are being dropped - e.g. in https://github.com/obophenotype/CCF_tools/issues/111#issuecomment-980091830


1. Some terms in the first graph are singletons but should be wired up, e.g.  
![image](https://user-images.githubusercontent.com/112839/143609688-fdbb7f53-49c1-4f46-bdf1-6c80ff12e61a.png)
should have a part_of relationship to 'kidney interstitum' - these terms are missing from graph 2
3. Some other edges missing, e.g. 'renal efferent arteriole' should be under 'kidney vasculature' and under 'renal corpuscle'

Reply:
Potential fix:
1. SELECT all triples from redundant graph with part_of in subject position and one of allowed OPs as predicate = set S
1. SELECT all triples from redundant graph with part_of in object position and one of allowed OPs as predicate = set  O
2. Generate intersection of sets S & O and use these to generate triples with SPARQL construct (as now)